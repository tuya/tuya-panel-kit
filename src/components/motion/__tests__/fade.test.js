import React from 'react';
import { View } from 'react-native';
import renderer from 'react-test-renderer';
import Motion from '../index';

describe('Motion Render', () => {
  it('basic render', () => {
    const component = renderer
      .create(
        <Motion.Fade show={true} onHide={jest.fn()}>
          <View style={{ width: 120, height: 48 }} />
        </Motion.Fade>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('invisible render', () => {
    const component = renderer
      .create(
        <Motion.Fade show={false} onHide={jest.fn()}>
          <View style={{ width: 120, height: 48 }} />
        </Motion.Fade>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('onHide render', () => {
    const onHide = jest.fn();
    jest.useFakeTimers();
    const component = renderer.create(
      <Motion.Fade show={true} onHide={onHide}>
        <View style={{ width: 120, height: 48 }} />
      </Motion.Fade>
    );
    expect(component).toMatchSnapshot();
    component.update(
      <Motion.Fade show={false} onHide={onHide}>
        <View style={{ width: 120, height: 48 }} />
      </Motion.Fade>
    );
    jest.runAllTimers();
    expect(onHide).toHaveBeenCalled();
    expect(component).toMatchSnapshot();
  });

  it('onShow render', () => {
    const onShow = jest.fn();
    jest.useFakeTimers();
    const component = renderer.create(
      <Motion.Fade show={false} onShow={onShow}>
        <View style={{ width: 120, height: 48 }} />
      </Motion.Fade>
    );
    expect(component).toMatchSnapshot();
    component.update(
      <Motion.Fade show={true} onShow={onShow}>
        <View style={{ width: 120, height: 48 }} />
      </Motion.Fade>
    );
    jest.runAllTimers();
    expect(onShow).toHaveBeenCalled();
    component.unmount();
    expect(component).toMatchSnapshot();
  });
});
