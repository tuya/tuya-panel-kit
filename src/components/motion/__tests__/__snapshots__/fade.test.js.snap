// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Motion Render basic render 1`] = `
<View
  renderToHardwareTextureAndroid={false}
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <View
    style={
      Object {
        "height": 48,
        "width": 120,
      }
    }
  />
</View>
`;

exports[`Motion Render invisible render 1`] = `null`;

exports[`Motion Render onHide render 1`] = `
<View
  renderToHardwareTextureAndroid={false}
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <View
    style={
      Object {
        "height": 48,
        "width": 120,
      }
    }
  />
</View>
`;

exports[`Motion Render onHide render 2`] = `null`;

exports[`Motion Render onShow render 1`] = `null`;

exports[`Motion Render onShow render 2`] = `null`;
